src/cpp/fastdds/domain/DomainParticipantFactory.cpp

```diff
@@ -332,10 +332,7 @@ ReturnCode_t DomainParticipantFactory::load_profiles()
         XMLProfileManager::loadDefaultXMLFile();
         default_xml_profiles_loaded = true;
 
-        if (default_participant_qos_ == PARTICIPANT_QOS_DEFAULT)
-        {
-            reset_default_participant_qos();
-        }
+        reset_default_participant_qos();
```

Comment:
@MiguelCompany :
Eduardo and I consider that condition undesired. May prevent a programmatic multiple load from different files to work properly. What do you think?

Reply:
Not only programmatic, the environment variable could be change to some other file outside the programs as well

Reply:
If the user has called `set_default_participant_qos`, calling `load_profiles` should not modify `default_participant_qos_`

Reply:
A comment informing of why the check is done would be welcome. Something like `// Only change default participant qos when not explicitly set by the user`